rollastic/node.py

```diff
@@ -54,13 +55,26 @@ def populate(self):
         self.clear()
 
         info = self.cluster.es.nodes.info(self.node_id)['nodes'].get(self.node_id, {})
+        isbad = False
```

Comment:
The lesson really is about ensuring that we maintain readability for the ideal case. Someone should be able to walk into this function and read it top to bottom and understand easily what it's trying to do and understand how it can mess up. Having flag variables / dirty bits / evolving conditional variables both disrupts the flow and is harder to reason about.